### PR TITLE
Fix NPE when editing OAuth2 applications

### LINKED ISSUE #27072: 500 Error when editing OAuth2 applications
### Description

Steps to reproduce
1. Click Edit

2. Get 500 Error

### Gitea Version

main

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

command line

### Database

None

### PR DESCRIPTION
**Pull request by JakobDev**
Fixes #27072

It looks like there are some cases where `ContextUser` is not set here

### CODE REVIEW

File: routers/web/user/setting/oauth2_common.go [hunk 1]
```diff
@@ -27,11 +27,13 @@ func (oa *OAuth2CommonHandlers) renderEditPage(ctx *context.Context) {
...
-			return
+	if ctx.ContextUser != nil {
+		if ctx.ContextUser.IsOrganization() {
+			err := shared_user.LoadHeaderCount(ctx)
+			if err != nil {
+				ctx.ServerError("LoadHeaderCount", err)
+				return
+			}
 		}
 	}
```
Comment (delvh) [inline routers/web/user/setting/oauth2_common.go]:
```suggestion
	if ctx.ContextUser == nil {
		return ctx.HTML(http.StatusOK, oa.TplAppEdit)
	}
	if ctx.ContextUser.IsOrganization() {
		err := shared_user.LoadHeaderCount(ctx)
		if err != nil {
			ctx.ServerError("LoadHeaderCount", err)
			return
		}
	}
```

Comment (wxiaoguang) [reply routers/web/user/setting/oauth2_common.go]:
```suggestion
	if ctx.ContextUser != nil && ctx.ContextUser.IsOrganization() {
		if err := shared_user.LoadHeaderCount(ctx); err != nil {
			ctx.ServerError("LoadHeaderCount", err)
			return
		}
	}
```

Or just this?

**Review by delvh — APPROVED** [review]

**Comment by lafriks** [comment]
If there is no context user how can we be sure we can even show/render this page? This would be security issue

**Comment by JakobDev** [comment]
The missing `ContextUser` happens when trying to create a new OAuth2 App for Users which is done using a POST request. I'm not sure if there are other places this could happen.

**Comment by wxiaoguang** [comment]
> If there is no context user how can we be sure we can even show/render this page? This would be security issue

What do you mean by "no context user" or "security issue"? This is shared code. When you are managing system oauth2 applications, there is no current context user.

**Comment by lafriks** [comment]
> > If there is no context user how can we be sure we can even show/render this page? This would be security issue
> 
> What do you mean by "no context user" or "security issue"? This is shared code. When you are managing system oauth2 applications, there is no current context user.

Oh. ok I somehow thought about context user as authorized user, all good sorry ;)

**Comment by wxiaoguang** [comment]
> Oh. ok I somehow thought about context user as authorized user, all good sorry ;)

`authorized user` is `ctx.Doer`, not `ctx.ContextUser`

**Comment by lafriks** [comment]
I know, it was late for me and somehow I mixed them up 😅

**Review by lafriks — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**